.buildkite/pipeline.yml

```diff
@@ -3,7 +3,6 @@
 env:
   SETUP_GVM_VERSION: 'v0.5.0'
   DOCKER_COMPOSE_VERSION: '1.25.5'
-  DOCKER_REGISTRY: "docker.elastic.co"
   DOCKER_IMAGE: "${DOCKER_REGISTRY}/observability-ci/temp-fleet-server" # needs to rename after tests
   DOCKER_IMAGE_TAG: ""
```

Comment:
I cannot add the message to the corresponding line, adding it here.

About `DOCKER_IMAGE_GIT_TAG` it is used `BUILDKITE_BRANCH`, but I see there is a variable in buildkite for tags `BUILDKITE_TAG`. What is the difference ? Should we use `BUILDKITE_TAG` here? 

```
DOCKER_IMAGE_GIT_TAG: "temp-${BUILDKITE_TAG}" # needs to rename after tests
```

Reply:
It looks like they are both set in case of pushing a tag ([link](https://buildkite.com/elastic/elastic-package/builds/748#01881e9e-15a3-4ee9-809d-8cd01598cf47)):
```
BUILDKITE_TAG="v0.80.0"
BUILDKITE_BRANCH="v0.80.0"
```

And for a push to main branch ([link](https://buildkite.com/elastic/elastic-package/builds/779#01884815-cb47-4b4c-bb95-2b39cbe79978)):
```
BUILDKITE_BRANCH="main"
BUILDKITE_TAG=""
```

Should we use `BUILDKITE_TAG` ?

Reply:
I just used the same logic:
https://github.com/elastic/fleet-server/blob/3229ead9040bc7f9c7bb1e6e7cc2730ab8f1dabf/.ci/Jenkinsfile#L140-L142
I can't find the output of the pipeline with values of these variables, because it looks like the Buildkite stores limited history.

Reply:
Jenkins stores the in BRANCH_NAME branch or tag. 

I guess looking at the output of BUILDKITE env. vars it could be used BUILDKITE_BRANCH too. However, as this step is intended to create docker tags according to the git tags pushed (or latest), I would use BUILDKITE_TAG environment variable here. Even that variable is the one used in the if clause.

Reply:
I guess for now we can leave it be the `branch_name`. If this is causing an issue we can do a follow up.